Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Bugzilla44166.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Xamarin.Forms.CustomAttributes;
+using Xamarin.Forms.Internals;
+
+#if UITEST
+using Xamarin.UITest;
+using NUnit.Framework;
+#endif
+
+namespace Xamarin.Forms.Controls
+{
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Bugzilla, 44166, "MasterDetailPage instances do not get disposed upon GC")]
+	public class Bugzilla44166 : TestContentPage
+	{
+		protected override void Init()
+		{
+			var label = new Label() { Text = "Testing..." };
+
+			var goButton = new Button { Text = "Go" };
+			goButton.Clicked += (sender, args) => Application.Current.MainPage = new _44166MDP();
+
+			var gcButton = new Button { Text = "GC" };
+			gcButton.Clicked += (sender, args) =>
+			{
+				GC.Collect();
+				GC.WaitForPendingFinalizers();
+
+				if (_44166MDP.Counter > 0)
+				{
+					Debug.WriteLine($">>>>>>>> Post-GC, {_44166MDP.Counter} {nameof(_44166MDP)} allocated");
+				}
+
+				if (_44166Master.Counter > 0)
+				{
+					Debug.WriteLine($">>>>>>>> Post-GC, {_44166Master.Counter} {nameof(_44166Master)} allocated");
+				}
+
+				if (_44166Detail.Counter > 0)
+				{
+					Debug.WriteLine($">>>>>>>> Post-GC, {_44166Detail.Counter} {nameof(_44166Detail)} allocated");
+				}
+
+				if (_44166NavContent.Counter > 0)
+				{
+					Debug.WriteLine($">>>>>>>> Post-GC, {_44166NavContent.Counter} {nameof(_44166NavContent)} allocated");
+				}
+
+				if (_44166NavContent.Counter + _44166Detail.Counter + _44166Master.Counter + _44166MDP.Counter == 0)
+				{
+					label.Text = "Success";
+				}
+			};
+
+			Content = new StackLayout
+			{
+				Children = { label, goButton, gcButton }
+			};
+		}
+
+		#if UITEST
+		[Test]
+		public void Bugzilla44166Test()
+		{
+			RunningApp.WaitForElement(q => q.Marked("Go"));
+			RunningApp.Tap(q => q.Marked("Go"));
+
+			RunningApp.WaitForElement(q => q.Marked("Back"));
+			RunningApp.Tap(q => q.Marked("Back"));
+
+			for(int n = 0; n < 10; n++)
```

Comment:
If you happen to need to make a change for this PR, you may want to add a space here after the `for`. :smile: